### PR TITLE
[Test] Create UnitTest and FuncionalTest. #19414.

### PR DESCRIPTION
**Pull request by amatosc**

### CODE REVIEW

**Review by demonodojo — APPROVED** [review]

File: sinfonier-web/test/tests/controllers/DiagnosisTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+public class DiagnosisTest extends BaseTestFunctional {
+
+  private static Http.Request request;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    doMongoImport(TestData.USERS_COLLECTION, TestData.USERS_JSON_FILE);
+    doMongoImport(TestData.MODULE_VERSIONS_COLLECTION, TestData.MODULE_VERSIONS_JSON_FILE);
+    doMongoImport(TestData.MODULES_COLLECTION, TestData.MODULES_JSON_FILE);
+    request = doLogin("test@test.com", "test");
```
Comment (fandrod) [inline sinfonier-web/test/tests/controllers/DiagnosisTest.java]: Diagnosis test must be done without user logged in

**Review by fandrod — CHANGES_REQUESTED** [review]

**Review by fandrod — COMMENTED** [review]
Why some tests class have "doLogin" call on @BeforeClass and others on @Before? If it is possible do several test with same user session, I think it is better on @BeforeClass

File: sinfonier-web/test/tests/controllers/DiagnosisTest.java (hunk 1 above)
Comment (amatosc) [reply sinfonier-web/test/tests/controllers/DiagnosisTest.java]: It's right, I will change it now.

**Comment by amatosc** [comment]
I did in this way because the request could be modified by the frameworks and it cause some troubles. Anyway I tried it in the first place but I had to change for fixing

**Review by fandrod — APPROVED** [review]

File: sinfonier-web/test/tests/controllers/ProfileSinfonierTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    doMongoImport(TestData.USERS_COLLECTION, TestData.USERS_JSON_FILE);
+    doMongoImport(TestData.MODULE_VERSIONS_COLLECTION, TestData.MODULE_VERSIONS_JSON_FILE);
+    doMongoImport(TestData.MODULES_COLLECTION, TestData.MODULES_JSON_FILE);
+  }
+
+  @Test
+  public void index() {
+    assertTrue(true);
```
Comment (ivan11paths) [inline sinfonier-web/test/tests/controllers/ProfileSinfonierTest.java]: This test isn't functional

**Review by ivan11paths — CHANGES_REQUESTED** [review]

**Review by ivan11paths — APPROVED** [review]

**Pull request merged**